hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -3265,6 +3265,7 @@ private TableGroup createTableGroup(TableGroup parentTableGroup, SqmPath<?> join
 							false,
 							this
 					);
+					tableGroupJoin.setImplicit();
```

Comment:
I think it should be possible within `createTableGroupJoin` to determine whether a join is implicit, by checking if `sqlAstJoinType == null`, so I'd prefer to remove `setImplicit` and instead make the field final in `TableGroupJoin`.